src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -166,33 +168,48 @@ public void AddPeer(NetworkPeer peer)
             this.ConnectedPeers.Add(peer);
         }
 
+        /// <summary>Determines whether or not a connector can be started.</summary>
+        internal abstract bool OnCanStartConnectAsync { get; }
```

Comment:
I would rename this `CanStartConnect` no need for async its not returning a task.

Reply:
Renamed